controllers/manager/gatewayvmconfiguration_controller.go

```diff
@@ -560,13 +560,62 @@ func (r *GatewayVMConfigurationReconciler) reconcileVMSSVM(
 		return "", fmt.Errorf("vmss vm(%s) has empty os profile", to.Val(vm.InstanceID))
 	}
 
+	forceUpdate := false
+	// check LatestModelApplied
+	if vm.Properties.LatestModelApplied != nil && !to.Val(vm.Properties.LatestModelApplied) {
+		forceUpdate = true
+		log.Info("Force update for unexpected VMSS instance LatestModelApplied:false", "vmssInstance", to.Val(vm.ID))
+	}
+
+	// check ProvisioningState
+	if !forceUpdate && vm.Properties.ProvisioningState != nil && !strings.EqualFold(to.Val(vm.Properties.ProvisioningState), "Succeeded") {
+		log.Info(fmt.Sprintf("VMSS instance ProvisioningState %q", to.Val(vm.Properties.ProvisioningState)))
+		if strings.EqualFold(to.Val(vm.Properties.ProvisioningState), "Failed") {
```

Comment:
could you add a test for ProvisioningState==Failed scenario?

Reply:
added to log by `log := log.FromContext(ctx).WithValues("vmssInstance", to.Val(vm.ID), "wantIPConfig", wantIPConfig, "ipPrefixID", ipPrefixID)`